lib/stdlib/src/lists.erl

```diff
@@ -391,6 +392,26 @@ delete(_, []) -> [].
 zip([X | Xs], [Y | Ys]) -> [{X, Y} | zip(Xs, Ys)];
 zip([], []) -> [].
 
+%% Return [{A0, ..., Z0}, ..., {An, ..., Zn}] for lists [A0, ..., An],
+%% [B0, ..., Bn], ..., [Z0, ..., Zn].
+
+-spec rotate(Lists) -> Lists when
+      Lists :: [[A]],
+      A :: term().
+
+rotate(Lists) -> rotate(Lists, []).
+
+rotate([[] | _], Acc) -> lists:reverse(Acc);
+
+rotate(Lists, Acc) ->
+    {Row, NewLists} = rotate_lists(Lists, {[], []}),
+    rotate(lists:reverse(NewLists), [lists:reverse(Row)|Acc]).
```

Comment:
@bjorng I'm not getting exactly what you propose

Reply:
My idea was to minimize the number of calls to `reverse/1`, but it turned out to be a little bit trickier than I naively thought it would be. If I'll get it to work and it's not too messy, I'll post the code.